### PR TITLE
Application

### PR DESCRIPTION
**Pull request by ujrlh**
I do have to fix one test tomorow, but apart from that everything should be fine.

### CODE REVIEW

File: src/osm_configurator/model/application/passive_project.py [hunk 1]
```diff
@@ -43,13 +47,13 @@ def get_edit_date(self):
...
-    def get_project_folder_path(self):
+    def get_project_folder_path(self) -> Path:
         """
         Gives back the path pointing towards the passive project.
 
         Returns:
             pathlib.Path: The path pointing towards the passive project.
         """
-        pass
+        return Path(self.data[2][1])
```
Comment (LuposX) [inline src/osm_configurator/model/application/passive_project.py:59]: magic number

File: src/osm_configurator/model/application/passive_project.py [hunk 2]
```diff
@@ -43,13 +47,13 @@ def get_edit_date(self):
         Returns:
             str: The last edit date of the passive project.
         """
-        pass
+        return self.data[4][1]
```
Comment (LuposX) [inline src/osm_configurator/model/application/passive_project.py:50]: magic number

File: src/osm_configurator/model/application/application.py [hunk 3]
```diff
@@ -10,97 +26,44 @@ def __init__(self):
...
-    def export_map(self, path):
-        pass
+        self.active_project: ActiveProject = None
+        self.application_settings: ApplicationSettings = ApplicationSettings()
+        self.passive_project_list: List[PassiveProject] = self._create_passive_project_list(
+            self.application_settings.get_default_location())
+        self.recommender_system: RecommenderSystem = RecommenderSystem()
+        self.application_settings_saver: ApplicationSettingsSaver = ApplicationSettingsSaver()
+
+    def create_project(self, name: str, description: str, destination: pathlib.Path) -> bool:
```
[the remark this answers is not in the document]
Comment (GloeStoe) [reply src/osm_configurator/model/application/application.py:36]: inheritance

File: src/osm_configurator/model/application/application_settings_saver.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        This method saves the default path where new projects should be stored.
+
+        Args:
+            destination (pathlib.Path): The path pointing towards the default project folder.
+
+        Returns:
+            bool: True when saving the path works, otherwise false.
+        """
+        if os.path.exists(destination):
+            filename = "default_project_folder.txt"
```
[the remark this answers is not in the document]
Comment (GloeStoe) [reply src/osm_configurator/model/application/application_settings_saver.py:33]: mybe don't use a .txt but a more descriptive extension

File: src/osm_configurator/model/application/application.py [hunk 5]
```diff
@@ -1,6 +1,22 @@
 from __future__ import annotations
 
+from pathlib import Path
+import os
+from typing import TYPE_CHECKING, List
+
 from src.osm_configurator.model.application.application_interface import IApplication
+from src.osm_configurator.model.application.application_settings import ApplicationSettings
```
Comment (GloeStoe) [inline src/osm_configurator/model/application/application.py:8]: don't import with from!

File: src/osm_configurator/model/application/application_interface.py [hunk 6]
```diff
@@ -29,20 +38,20 @@ def create_project(self, name, description, destination):
...
-            path (pathlib.Path): The path of the project, to be loaded.
+            destination (pathlib.Path): The path of the project, to be loaded.
         Returns:
             bool: True if loading the project is working, otherwise false.
         """
         pass
 
     @abstractmethod
-    def get_passive_project_list(self):
+    def get_passive_project_list(self) -> list[PassiveProject]:
```
Comment (GloeStoe) [inline src/osm_configurator/model/application/application_interface.py:54]: List is written big

File: src/osm_configurator/model/application/passive_project.py [hunk 7]
```diff
@@ -9,32 +11,34 @@ class PassiveProject:
     the class holds the name, description, last edit date and path of the projects.
     """
 
-    def __init__(self, project_folder_path):
+    def __init__(self, settings_file):
```
Comment (GloeStoe) [inline src/osm_configurator/model/application/passive_project.py:14]: typehints

File: src/osm_configurator/model/application/application_interface.py [hunk 8]
```diff
@@ -29,30 +38,30 @@ def create_project(self, name, description, destination):
         pass
 
     @abstractmethod
-    def load_project(self, path):
+    def load_project(self, destination) -> bool:
```
Comment (LuposX) [inline src/osm_configurator/model/application/application_interface.py:41]: typehint

File: src/osm_configurator/model/application/application_settings_io_handler.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            application_settings_file (Path): name of the file
+        """
+        self.filename: Path = application_settings_file
+
+    def load_settings_file(self) -> Path:
+        """
+        This method loads the settings file.
+
+        Returns:
+            bool: True when building the path works, otherwise false.
```
Comment (LuposX) [inline src/osm_configurator/model/application/application_settings_io_handler.py:31]: wron tpyrhint

File: src/osm_configurator/model/application/passive_project.py [hunk 10]
```diff
@@ -9,47 +12,49 @@ class PassiveProject:
...
         """
-        Gives back the path pointing towards the passive project.
+        Gives back the last edit date of the passive project.
 
         Returns:
-            pathlib.Path: The path pointing towards the passive project.
+            str: The last edit date of the passive project.
         """
-        pass
+        return self.data[4][1]
```
Comment (LuposX) [inline src/osm_configurator/model/application/passive_project.py:60]: still magic numbers.

**Review by LuposX — CHANGES_REQUESTED** [review]

**Review by LuposX — APPROVED** [review]

**Pull request merged**